### ISSUE TITLE
"get -r" nor wildcard does work in UberFTP 2.9 (but worked in 2.8!)

### ISSUE THREAD

**Issue by takudoi** [gridcf/UberFTP#20]
Using uberftp 2.9, I tried to download a directory recursively as
> get -r rdir wdir

However, I found that only empty directory "rdir" is created, and that is all.

I also tried to download multiple files using wildcard with mget, but it failed as follows.
> mget *
No match for *

I installed uberftp 2.8 and then both of "get -r" and wildcard worked.
So, it seems that these are bugs in 2.9.

**Comment by fscheiner** [comment]
Strange. Thanks for reporting, we'll have a look and report back.

**Comment by fscheiner** [comment]
@takudoi takudoi
I could reproduce the reported problems and hence can confirm this as a regression in UberFTP 2.9.

CCing to @msalle @maarten-litmaath @traylenator

Here's a log of the tested operations with 2.8 and 2.9 for reference. The server is a Globus GridFTP server 13.22 from GCT 6.2.20210826 on CentOS 7.9.

* with UberFTP 2.9 (uberftp-2.9-1.el7.x86_64 on CentOS 7.9):
```
[johndoe@client-1 uberftp-2.9]$ uberftp
UberFTP (2.9)> open gridftp-5.domain.tld
220 gridftp-5.domain.tld GridFTP Server 13.22 (gcc64, 1629915172-0) [Grid Community Toolkit 6.2] ready.
230 User johndoe2 logged in.
UberFTP (2.9)> ls
drwx------   6 johndoe2 johndoe2           74 Jul  7 15:29 .
drwxr-xr-x   4     root     root           37 Mar 11 14:04 ..
-rw-r--r--   1 johndoe2 johndoe2           18 Nov 24 17:33 .bash_logout
-rw-r--r--   1 johndoe2 johndoe2          193 Nov 24 17:33 .bash_profile
-rw-r--r--   1 johndoe2 johndoe2          231 Nov 24 17:33 .bashrc
drwxr-xr-x   2 johndoe2 johndoe2          106 Jul  7 15:31 rdir
UberFTP (2.9)> help get
Retreive file(s) from the remote service. If <source> implies multiple
transfers, either through regular expressions or by using the recursive
feature, then [destination] must be a directory. If [destination] is not
specified, <source> is used.

Usage: get [-r] <source> [destination]
-r   Recursively transfer the given directory.
UberFTP (2.9)> get -r rdir ./
UberFTP (2.9)> lls
total 0
drwxrwxr-x. 3 johndoe johndoe 18 Jul  7 15:33 .
drwxrwxr-x. 5 johndoe johndoe 51 Jul  7 15:29 ..
drwx------. 2 johndoe johndoe  6 Jul  7 15:33 rdir
UberFTP (2.9)> lcd rdir
UberFTP (2.9)> ls
drwx------   6 johndoe2 johndoe2           74 Jul  7 15:29 .
drwxr-xr-x   4     root     root           37 Mar 11 14:04 ..
-rw-r--r--   1 johndoe2 johndoe2           18 Nov 24 17:33 .bash_logout
-rw-r--r--   1 johndoe2 johndoe2          193 Nov 24 17:33 .bash_profile
-rw-r--r--   1 johndoe2 johndoe2          231 Nov 24 17:33 .bashrc
drwxr-xr-x   2 johndoe2 johndoe2          106 Jul  7 15:31 rdir
UberFTP (2.9)> lls
total 0
drwx------. 2 johndoe johndoe  6 Jul  7 15:33 .
drwxrwxr-x. 3 johndoe johndoe 18 Jul  7 15:33 ..
UberFTP (2.9)> cd rdir
UberFTP (2.9)> ls
drwxr-xr-x  12 johndoe2 johndoe2          106 Jul  7 15:31 .
drwx------   3 johndoe2 johndoe2           74 Jul  7 15:29 ..
-rw-r--r--   1 johndoe2 johndoe2      7497216 Jul  7 15:31 01
-rw-r--r--   1 johndoe2 johndoe2      3644672 Jul  7 15:31 02
-rw-r--r--   1 johndoe2 johndoe2      3093248 Jul  7 15:31 03
-rw-r--r--   1 johndoe2 johndoe2      4335360 Jul  7 15:31 04
-rw-r--r--   1 johndoe2 johndoe2      3384064 Jul  7 15:31 05
-rw-r--r--   1 johndoe2 johndoe2      1332480 Jul  7 15:31 06
-rw-r--r--   1 johndoe2 johndoe2      6842112 Jul  7 15:31 07
-rw-r--r--   1 johndoe2 johndoe2      6880768 Jul  7 15:31 08
-rw-r--r--   1 johndoe2 johndoe2      2800640 Jul  7 15:31 09
-rw-r--r--   1 johndoe2 johndoe2      5315840 Jul  7 15:31 10
UberFTP (2.9)> help mget
Retrieve file(s) from the remote service. This is similiar to making
multiple calls to get without specifying a destination.

Usage: mget [-r] object1 [object2...objectn]
-r   Recursively transfer the given directory.
UberFTP (2.9)> mget *
No match for *
UberFTP (2.9)> mget -r *
No match for *
UberFTP (2.9)> mget \* 
No match for *
UberFTP (2.9)>

[johndoe@client-1 uberftp-2.9]$ uberftp gsiftp://gridftp-5.domain.tld/~/rdir file:///$PWD/
[johndoe@client-1 uberftp-2.9]$ ls
rdir
[johndoe@client-1 uberftp-2.9]$ ls rdir
[johndoe@client-1 uberftp-2.9]$ uberftp gsiftp://gridftp-5.domain.tld/~/rdir/* file:///$PWD/
No match for ~/rdir/*
```

* with UberFTP 2.8 (uberftp-2.8-13.el8.x86_64 on RockyLinux 8.6):
```
[johndoe@client-2 ubeftp-2.8]$ uberftp
UberFTP (2.8)> open gridftp-5.domain.tld
220 gridftp-5.domain.tld GridFTP Server 13.22 (gcc64, 1629915172-0) [Grid Community Toolkit 6.2] ready.
230 User johndoe2 logged in.
UberFTP (2.8)> ls
drwx------   6 johndoe2 johndoe2           74 Jul  7 15:29 .
drwxr-xr-x   4     root     root           37 Mar 11 14:04 ..
-rw-r--r--   1 johndoe2 johndoe2           18 Nov 24 17:33 .bash_logout
-rw-r--r--   1 johndoe2 johndoe2          193 Nov 24 17:33 .bash_profile
-rw-r--r--   1 johndoe2 johndoe2          231 Nov 24 17:33 .bashrc
drwxr-xr-x   2 johndoe2 johndoe2          106 Jul  7 15:31 rdir
UberFTP (2.8)> get -r rdir
rdir/01: 7497216 bytes in 0.078925 Seconds (90.591 MB/s)
rdir/02: 3644672 bytes in 0.034037 Seconds (102.119 MB/s)
rdir/03: 3093248 bytes in 0.040009 Seconds (73.732 MB/s)
rdir/04: 4335360 bytes in 0.044925 Seconds (92.032 MB/s)
rdir/05: 3384064 bytes in 0.036753 Seconds (87.810 MB/s)
rdir/06: 1332480 bytes in 0.028106 Seconds (45.213 MB/s)
rdir/07: 6842112 bytes in 0.062485 Seconds (104.427 MB/s)
rdir/08: 6880768 bytes in 0.051719 Seconds (126.878 MB/s)
rdir/09: 2800640 bytes in 0.040784 Seconds (65.489 MB/s)
rdir/10: 5315840 bytes in 0.057983 Seconds (87.432 MB/s)
UberFTP (2.8)> lls
total 0
drwxrwxr-x. 3 johndoe johndoe  18 Jul  7 15:41 .
drwxrwxr-x. 3 johndoe johndoe  24 Jul  7 15:40 ..
drwx------. 2 johndoe johndoe 106 Jul  7 15:41 rdir
UberFTP (2.8)> cd rdir         
UberFTP (2.8)> ls    
drwxr-xr-x  12 johndoe2 johndoe2          106 Jul  7 15:31 .
drwx------   3 johndoe2 johndoe2           74 Jul  7 15:29 ..
-rw-r--r--   1 johndoe2 johndoe2      7497216 Jul  7 15:31 01
-rw-r--r--   1 johndoe2 johndoe2      3644672 Jul  7 15:31 02
-rw-r--r--   1 johndoe2 johndoe2      3093248 Jul  7 15:31 03
-rw-r--r--   1 johndoe2 johndoe2      4335360 Jul  7 15:31 04
-rw-r--r--   1 johndoe2 johndoe2      3384064 Jul  7 15:31 05
-rw-r--r--   1 johndoe2 johndoe2      1332480 Jul  7 15:31 06
-rw-r--r--   1 johndoe2 johndoe2      6842112 Jul  7 15:31 07
-rw-r--r--   1 johndoe2 johndoe2      6880768 Jul  7 15:31 08
-rw-r--r--   1 johndoe2 johndoe2      2800640 Jul  7 15:31 09
-rw-r--r--   1 johndoe2 johndoe2      5315840 Jul  7 15:31 10
UberFTP (2.8)> mget *
01: 7497216 bytes in 0.071962 Seconds (99.357 MB/s)
02: 3644672 bytes in 0.039971 Seconds (86.959 MB/s)
03: 3093248 bytes in 0.040882 Seconds (72.158 MB/s)
04: 4335360 bytes in 0.041291 Seconds (100.131 MB/s)
05: 3384064 bytes in 0.040597 Seconds (79.496 MB/s)
06: 1332480 bytes in 0.029117 Seconds (43.643 MB/s)
07: 6842112 bytes in 0.050648 Seconds (128.833 MB/s)
08: 6880768 bytes in 0.051965 Seconds (126.278 MB/s)
09: 2800640 bytes in 0.033682 Seconds (79.298 MB/s)
10: 5315840 bytes in 0.046360 Seconds (109.352 MB/s)
UberFTP (2.8)> lls
total 44088
drwxrwxr-x. 3 johndoe johndoe     118 Jul  7 15:42 .
drwxrwxr-x. 3 johndoe johndoe      24 Jul  7 15:40 ..
-rw-------. 1 johndoe johndoe 7497216 Jul  7 15:31 01
-rw-------. 1 johndoe johndoe 3644672 Jul  7 15:31 02
-rw-------. 1 johndoe johndoe 3093248 Jul  7 15:31 03
-rw-------. 1 johndoe johndoe 4335360 Jul  7 15:31 04
-rw-------. 1 johndoe johndoe 3384064 Jul  7 15:31 05
-rw-------. 1 johndoe johndoe 1332480 Jul  7 15:31 06
-rw-------. 1 johndoe johndoe 6842112 Jul  7 15:31 07
-rw-------. 1 johndoe johndoe 6880768 Jul  7 15:31 08
-rw-------. 1 johndoe johndoe 2800640 Jul  7 15:31 09
-rw-------. 1 johndoe johndoe 5315840 Jul  7 15:31 10
drwx------. 2 johndoe johndoe     106 Jul  7 15:41 rdir
UberFTP (2.8)>

[johndoe@client-2 ubeftp-2.8]$ uberftp gsiftp://gridftp-5.domain.tld/~/rdir file:///$PWD/
[johndoe@client-2 ubeftp-2.8]$ ls
rdir
[johndoe@client-2 ubeftp-2.8]$ ls rdir
[johndoe@client-2 ubeftp-2.8]$ uberftp gsiftp://gridftp-5.domain.tld/~/rdir/* file:///$PWD/
[johndoe@client-2 ubeftp-2.8]$ ls
01  02  03  04  05  06  07  08  09  10  rdir
```

**Comment by fscheiner** [comment]
I'll try without https://github.com/gridcf/UberFTP/commit/eb37d55484416634f7603c376246880ad94e0fea for a start.

**Comment by fscheiner** [comment]
https://github.com/gridcf/UberFTP/commit/eb37d55484416634f7603c376246880ad94e0fea is indeed the culprit, without it, it works as in UberFTP 2.8:

* with UberFTP 2.9 (w/o `eb37d55` on CentOS 7.9):
```
[johndoe@client-1 bin]$ ./uberftp 
UberFTP (2.9)> open gridftp-5.domain.tld
220 gridftp-5.domain.tld GridFTP Server 13.24 (gcc64, 1653033972-1) [Grid Community Toolkit 6.2] ready.
230 User johndoe2 logged in.
UberFTP (2.9)> pwd
/home/johndoe2
UberFTP (2.9)> ls
drwx------   6 johndoe2 johndoe2           74 Jul  7 15:29 .
drwxr-xr-x   4     root     root           37 Mar 11 14:04 ..
-rw-r--r--   1 johndoe2 johndoe2           18 Nov 24 17:33 .bash_logout
-rw-r--r--   1 johndoe2 johndoe2          193 Nov 24 17:33 .bash_profile
-rw-r--r--   1 johndoe2 johndoe2          231 Nov 24 17:33 .bashrc
drwxr-xr-x   2 johndoe2 johndoe2          106 Jul  7 15:31 rdir
UberFTP (2.9)> get -r rdir
rdir/01: 7497216 bytes in 0.079966 Seconds (89.412 MB/s)
rdir/02: 3644672 bytes in 0.040928 Seconds (84.925 MB/s)
rdir/03: 3093248 bytes in 0.050412 Seconds (58.517 MB/s)
rdir/04: 4335360 bytes in 0.047348 Seconds (87.322 MB/s)
rdir/05: 3384064 bytes in 0.041974 Seconds (76.888 MB/s)
rdir/06: 1332480 bytes in 0.028683 Seconds (44.303 MB/s)
rdir/07: 6842112 bytes in 0.058785 Seconds (111.000 MB/s)
rdir/08: 6880768 bytes in 0.062048 Seconds (105.757 MB/s)
rdir/09: 2800640 bytes in 0.047694 Seconds (56.001 MB/s)
rdir/10: 5315840 bytes in 0.054007 Seconds (93.869 MB/s)
UberFTP (2.9)> lls
total 0
drwx------. 3 johndoe johndoe  18 Jul  7 16:35 .
drwxrwxr-x. 6 johndoe johndoe  74 Jul  7 16:35 ..
drwx------. 2 johndoe johndoe 106 Jul  7 16:35 rdir
UberFTP (2.9)> lls rdir
total 44088
drwx------. 2 johndoe johndoe     106 Jul  7 16:35 .
drwx------. 3 johndoe johndoe      18 Jul  7 16:35 ..
-rw-------. 1 johndoe johndoe 7497216 Jul  7 15:31 01
-rw-------. 1 johndoe johndoe 3644672 Jul  7 15:31 02
-rw-------. 1 johndoe johndoe 3093248 Jul  7 15:31 03
-rw-------. 1 johndoe johndoe 4335360 Jul  7 15:31 04
-rw-------. 1 johndoe johndoe 3384064 Jul  7 15:31 05
-rw-------. 1 johndoe johndoe 1332480 Jul  7 15:31 06
-rw-------. 1 johndoe johndoe 6842112 Jul  7 15:31 07
-rw-------. 1 johndoe johndoe 6880768 Jul  7 15:31 08
-rw-------. 1 johndoe johndoe 2800640 Jul  7 15:31 09
-rw-------. 1 johndoe johndoe 5315840 Jul  7 15:31 10
UberFTP (2.9)> cd rdir
UberFTP (2.9)> ls
drwxr-xr-x  12 johndoe2 johndoe2          106 Jul  7 15:31 .
drwx------   3 johndoe2 johndoe2           74 Jul  7 15:29 ..
-rw-r--r--   1 johndoe2 johndoe2      7497216 Jul  7 15:31 01
-rw-r--r--   1 johndoe2 johndoe2      3644672 Jul  7 15:31 02
-rw-r--r--   1 johndoe2 johndoe2      3093248 Jul  7 15:31 03
-rw-r--r--   1 johndoe2 johndoe2      4335360 Jul  7 15:31 04
-rw-r--r--   1 johndoe2 johndoe2      3384064 Jul  7 15:31 05
-rw-r--r--   1 johndoe2 johndoe2      1332480 Jul  7 15:31 06
-rw-r--r--   1 johndoe2 johndoe2      6842112 Jul  7 15:31 07
-rw-r--r--   1 johndoe2 johndoe2      6880768 Jul  7 15:31 08
-rw-r--r--   1 johndoe2 johndoe2      2800640 Jul  7 15:31 09
-rw-r--r--   1 johndoe2 johndoe2      5315840 Jul  7 15:31 10
UberFTP (2.9)> mget *
01: 7497216 bytes in 0.078035 Seconds (91.624 MB/s)
02: 3644672 bytes in 0.043735 Seconds (79.475 MB/s)
03: 3093248 bytes in 0.040763 Seconds (72.368 MB/s)
04: 4335360 bytes in 0.053456 Seconds (77.344 MB/s)
05: 3384064 bytes in 0.044142 Seconds (73.112 MB/s)
06: 1332480 bytes in 0.028095 Seconds (45.231 MB/s)
07: 6842112 bytes in 0.055629 Seconds (117.298 MB/s)
08: 6880768 bytes in 0.056344 Seconds (116.463 MB/s)
09: 2800640 bytes in 0.037459 Seconds (71.302 MB/s)
10: 5315840 bytes in 0.055908 Seconds (90.677 MB/s)
UberFTP (2.9)> lls
total 44088
drwx------. 3 johndoe johndoe     118 Jul  7 16:36 .
drwxrwxr-x. 6 johndoe johndoe      74 Jul  7 16:35 ..
-rw-------. 1 johndoe johndoe 7497216 Jul  7 15:31 01
-rw-------. 1 johndoe johndoe 3644672 Jul  7 15:31 02
-rw-------. 1 johndoe johndoe 3093248 Jul  7 15:31 03
-rw-------. 1 johndoe johndoe 4335360 Jul  7 15:31 04
-rw-------. 1 johndoe johndoe 3384064 Jul  7 15:31 05
-rw-------. 1 johndoe johndoe 1332480 Jul  7 15:31 06
-rw-------. 1 johndoe johndoe 6842112 Jul  7 15:31 07
-rw-------. 1 johndoe johndoe 6880768 Jul  7 15:31 08
-rw-------. 1 johndoe johndoe 2800640 Jul  7 15:31 09
-rw-------. 1 johndoe johndoe 5315840 Jul  7 15:31 10
drwx------. 2 johndoe johndoe     106 Jul  7 16:35 rdir

[johndoe@client-1 uberftp-2.9-mod]$ ~/git-projects/UberFTP/bin/uberftp gsiftp://gridftp-5.domain.tld/~/rdir file:///$PWD/
[johndoe@client-1 uberftp-2.9-mod]$ ls
rdir
[johndoe@client-1 uberftp-2.9-mod]$ ls rdir
[johndoe@client-1 uberftp-2.9-mod]$ ~/git-projects/UberFTP/bin/uberftp gsiftp://gridftp-5.domain.tld/~/rdir/* file:///$PWD/
[johndoe@client-1 uberftp-2.9-mod]$ ls
01  02  03  04  05  06  07  08  09  10  rdir
```

**Comment by fscheiner** [comment]
The attached patch reverts https://github.com/gridcf/UberFTP/commit/eb37d55484416634f7603c376246880ad94e0fea.
[fix-recursive-operations-in-uberftp-2.9.patch.txt](https://github.com/gridcf/UberFTP/files/9065408/fix-recursive-operations-in-uberftp-2.9.patch.txt)

**Comment by fscheiner** [comment]
The actual problem lies in [`_f_readdir_mlsd()`](https://github.com/gridcf/UberFTP/blob/Version_2_9/ftp.c#L3993) missing to set `eol` to the end of the string `listing`. In [`_f_readdir_nlst()`](https://github.com/gridcf/UberFTP/blob/Version_2_9/ftp.c#L4105) it is done correctly:

https://github.com/gridcf/UberFTP/blob/c687176a0707b83cc3693c42c42a6e847f87e63a/ftp.c#L4201-L4202

...so this definition also needs to be added to `_f_readdir_mlsd()`.

**Comment by fscheiner** [comment]
Adding this definition solves this issue and makes `get -r` and `mget` work again:
```
[johndoe@client-1 uberftp-2.9-mod2]$ ~/git-projects/UberFTP/uberftp 
UberFTP (2.9)> open gridftp-5.domain.tld
220 gridftp-5.domain.tld GridFTP Server 13.24 (gcc64, 1653033972-1) [Grid Community Toolkit 6.2] ready.
230 User johndoe2 logged in.
UberFTP (2.9)> lls
total 0
drwxrwxr-x. 2 johndoe johndoe  6 Jul 14 16:29 .
drwxrwxr-x. 7 johndoe johndoe 98 Jul 14 15:20 ..
UberFTP (2.9)> ls
drwx------   6 johndoe2 johndoe2           74 Jul  7 15:29 .
drwxr-xr-x   4     root     root           37 Mar 11 14:04 ..
-rw-r--r--   1 johndoe2 johndoe2           18 Nov 24 17:33 .bash_logout
-rw-r--r--   1 johndoe2 johndoe2          193 Nov 24 17:33 .bash_profile
-rw-r--r--   1 johndoe2 johndoe2          231 Nov 24 17:33 .bashrc
drwxr-xr-x   2 johndoe2 johndoe2          106 Jul  7 15:31 rdir
UberFTP (2.9)> get -r rdir ./
rdir/01: 7497216 bytes in 0.077762 Seconds (91.946 MB/s)
rdir/02: 3644672 bytes in 0.040745 Seconds (85.307 MB/s)
rdir/03: 3093248 bytes in 0.037349 Seconds (78.983 MB/s)
rdir/04: 4335360 bytes in 0.051703 Seconds (79.967 MB/s)
rdir/05: 3384064 bytes in 0.046957 Seconds (68.729 MB/s)
rdir/06: 1332480 bytes in 0.032503 Seconds (39.096 MB/s)
rdir/07: 6842112 bytes in 0.066163 Seconds (98.622 MB/s)
rdir/08: 6880768 bytes in 0.063205 Seconds (103.821 MB/s)
rdir/09: 2800640 bytes in 0.039547 Seconds (67.537 MB/s)
rdir/10: 5315840 bytes in 0.060719 Seconds (83.492 MB/s)
UberFTP (2.9)> lls rdir
total 44088
drwx------. 2 johndoe johndoe     106 Jul 14 16:30 .
drwxrwxr-x. 3 johndoe johndoe      18 Jul 14 16:30 ..
-rw-------. 1 johndoe johndoe 7497216 Jul  7 15:31 01
-rw-------. 1 johndoe johndoe 3644672 Jul  7 15:31 02
-rw-------. 1 johndoe johndoe 3093248 Jul  7 15:31 03
-rw-------. 1 johndoe johndoe 4335360 Jul  7 15:31 04
-rw-------. 1 johndoe johndoe 3384064 Jul  7 15:31 05
-rw-------. 1 johndoe johndoe 1332480 Jul  7 15:31 06
-rw-------. 1 johndoe johndoe 6842112 Jul  7 15:31 07
-rw-------. 1 johndoe johndoe 6880768 Jul  7 15:31 08
-rw-------. 1 johndoe johndoe 2800640 Jul  7 15:31 09
-rw-------. 1 johndoe johndoe 5315840 Jul  7 15:31 10
UberFTP (2.9)> cd rdir
UberFTP (2.9)> mget *
01: 7497216 bytes in 0.076718 Seconds (93.197 MB/s)
02: 3644672 bytes in 0.044406 Seconds (78.274 MB/s)
03: 3093248 bytes in 0.050525 Seconds (58.386 MB/s)
04: 4335360 bytes in 0.053634 Seconds (77.088 MB/s)
05: 3384064 bytes in 0.051325 Seconds (62.880 MB/s)
06: 1332480 bytes in 0.030807 Seconds (41.249 MB/s)
07: 6842112 bytes in 0.064966 Seconds (100.439 MB/s)
08: 6880768 bytes in 0.062038 Seconds (105.774 MB/s)
09: 2800640 bytes in 0.041157 Seconds (64.895 MB/s)
10: 5315840 bytes in 0.050772 Seconds (99.850 MB/s)
UberFTP (2.9)> lls
total 44088
drwxrwxr-x. 3 johndoe johndoe     118 Jul 14 16:30 .
drwxrwxr-x. 7 johndoe johndoe      98 Jul 14 15:20 ..
-rw-------. 1 johndoe johndoe 7497216 Jul  7 15:31 01
-rw-------. 1 johndoe johndoe 3644672 Jul  7 15:31 02
-rw-------. 1 johndoe johndoe 3093248 Jul  7 15:31 03
-rw-------. 1 johndoe johndoe 4335360 Jul  7 15:31 04
-rw-------. 1 johndoe johndoe 3384064 Jul  7 15:31 05
-rw-------. 1 johndoe johndoe 1332480 Jul  7 15:31 06
-rw-------. 1 johndoe johndoe 6842112 Jul  7 15:31 07
-rw-------. 1 johndoe johndoe 6880768 Jul  7 15:31 08
-rw-------. 1 johndoe johndoe 2800640 Jul  7 15:31 09
-rw-------. 1 johndoe johndoe 5315840 Jul  7 15:31 10
drwx------. 2 johndoe johndoe     106 Jul 14 16:30 rdir
UberFTP (2.9)> 
```

**Comment by fscheiner** [comment]
Fixed with #21 but kept open until next release.

**Comment by fscheiner** [comment]
Fixed in [UberFTP 2.9.1 release](https://github.com/gridcf/UberFTP/releases/tag/Version_2_9_1).
